lib/driver.js

```diff
@@ -412,10 +412,20 @@ class XCUITestDriver extends BaseDriver {
       await this.wda.uninstall();
       this.logEvent('wdaUninstalled');
     } else if (!util.hasValue(this.wda.webDriverAgentUrl) && (await this.wda.isRunning())) {
```

Comment:
`this.wda.isRunning()` already checks the status. Perhaps it could be modified (with a name change, since it's functioning would be different) to also set things like the current bundleId for the running WDA?